disruption_py/core/utils/misc.py

```diff
@@ -132,3 +138,79 @@ def get_commit_hash() -> str:
     except subprocess.CalledProcessError:
         commit_hash = None
     return commit_hash
+
+
+@lru_cache
+def get_temporary_folder() -> Path:
+    """
+    Create and return a temporary folder.
+    The result is cached to return the same path for different invocations.
+
+    Returns
+    -------
+    Path
+        Resulting temporary folder.
+    """
+
+    # create temporary top folder
+    top = os.path.join("/tmp", os.getenv("USER"), "disruption-py")
+    Path(top).mkdir(parents=True, exist_ok=True)
+
+    # create temporary sub folder
+    return mkdtemp(prefix=time.strftime("%Y%m%d-%H%M%S-"), dir=top)
+
+
+def shot_log_msg(shot_id: int, message: str):
```

Comment:
I looked into it for a bit. `shot_log_msg` could return a dictionary 
```
{"_Logger__message": "#{shot_id} | {message}",
        "shot_id": shot_id, 
        "message": message}
``` 
but we would need to always unpack it: `logger.info(**shot_log_msg(shot_id, "this is a message")` which is annoying. 

I'll leave it as is because the lazy loading won't give a huge performance boost.